examples/run_torch_onnx.py

```diff
@@ -0,0 +1,163 @@
+
+import argparse
+import importlib
+import sys
+from pathlib import Path
+
+import cv2
+import numpy as np
+import openvino as ov
+import timm
+import torch
+
+from openvino_xai import Task, insert_xai
+from openvino_xai.common.utils import logger, softmax
+from openvino_xai.explainer.visualizer import colormap, overlay
+
+
+def get_argument_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--model_name", default="resnet18.a1_in1k", type=str)
+    parser.add_argument("--image_path", default="tests/assets/cheetah_person.jpg", type=str)
+    parser.add_argument("--output_dir", default=".data/example", type=str)
+    return parser
+
+
+def main(argv: list[str]):
+    parser = get_argument_parser()
+    args = parser.parse_args(argv)
+
+    # Load Torch model from timm
+    try:
+        model = timm.create_model(args.model_name, in_chans=3, pretrained=True)
+        logger.info(f"Model config: {model.default_cfg}")
+        logger.info(f"Model layers: {model}")
+    except Exception as e:
+        logger.error(e)
+        logger.info(f"Please choose from {timm.list_models()}")
+        sys.exit(1)
+    input_size = model.default_cfg["input_size"][1:]  # (H, W)
+    input_mean = np.array(model.default_cfg["mean"])
+    input_std = np.array(model.default_cfg["std"])
+
+    # Load image
+    image = cv2.imread("tests/assets/cheetah_person.jpg")
+    image = cv2.resize(image, dsize=input_size)
+    image = cv2.cvtColor(image, code=cv2.COLOR_BGR2RGB)
+    image_norm = ((image/255.0 - input_mean)/input_std).astype(np.float32)
+    image_norm = image_norm.transpose((2, 0, 1))  # HxWxC -> CxHxW
+    image_norm = image_norm[None, :]  # CxHxW -> 1xCxHxW
+
+    # Torch model inference
+    model.eval()
+    with torch.no_grad():
+        logits = model(torch.from_numpy(image_norm))
+        probs = torch.softmax(logits, dim=-1)  # BxC
+        label = probs.argmax(dim=-1)[0]
+    logger.info(f"Torch model prediction: classes ({probs.shape[-1]}) -> label ({label}) -> prob ({probs[0, label]})")
+
+    # Insert XAI head
+    xai_model: torch.nn.Module = insert_xai(model, Task.CLASSIFICATION)
+
+    # Torch XAI model inference
+    xai_model.eval()
+    with torch.no_grad():
+        outputs = xai_model(torch.from_numpy(image_norm))
+        logits = outputs["prediction"]  # BxC
+        saliency_maps = outputs["saliency_map"]  # BxCxhxw
+        probs = torch.softmax(logits, dim=-1)
+        label = probs.argmax(dim=-1)[0]
+    logger.info(f"Torch XAI model prediction: classes ({probs.shape[-1]}) -> label ({label}) -> prob ({probs[0, label]})")
+
+    # Torch XAI model saliency map
+    saliency_maps = saliency_maps.numpy(force=True).squeeze(0)  # Cxhxw
+    mask = saliency_maps[label]  # hxw mask for the label
+    mask = colormap(mask[None, :])  # 1xhxw
+    mask = cv2.resize(mask.squeeze(0), dsize=input_size)  # HxW
+    result = overlay(mask, image)
+    result = cv2.cvtColor(result, code=cv2.COLOR_RGB2BGR)
```

Comment:
"mask" -> "sal_map"
"result" -> "result_saliency_map"?

Reply:
Refined in https://github.com/openvinotoolkit/openvino_xai/pull/66/commits/44426944a60f027ab3978c9c5d4a45827e2c5aee

Reply:
One small update: "saliency_image" -> "overlayed_saliency_map" (Can be done later)

Reply:
Changed to `result_image` as we discussed.